### ISSUE TITLE
wdsr_b increase channel problem

### ISSUE THREAD

**Issue by suke27** [JiahuiYu/wdsr_ntire2018#26]
you make a increase channel unit in wdsr_b 32->32*6->32*0.8->32
I have a problem, does 32*6 is for scale 2 times, how about x3 and x4, should I change 6  to other number.
you paper said
 It enables even wider activation (6 to 9) without additional parameters or computation.

**Comment by suke27** [comment]
@JiahuiYu, I just try x6 case in x3 and x4 SR case, found the psnr is lower than EDSR baseline(16 res+64 filter), so I ask about it. by the way, x6-x9, is there linear scale between scale time and enlarge channel.
like x2 case enlarge x6, x3 case enlarge x9, x4 case enlarge x12?

**Comment by JiahuiYu** [comment]
You don't need to open new issue, let's make it clean so that others can also have a clear reference. Just reply here and I can see it.

Can you report your EDSR and WDSR in terms of (1) number of parameters, (2) channel number in residual identity path way, (3) channel number before activation, and (4) their final performance?

For x2SR, x3SR or x4SR, we use same x4 wider activation with WDSR-A, x6-x9 wider activation with WDSR-B. DO NOT use x6-x9 with WDSR-A, or others like x12, that will make your identity pathway too slim.

**Comment by suke27** [comment]
EDSR baseline 16 resblock filter 64, WDSR_A 16resblock filter 32->32*4->32, wdsr_b 16 resblock filter 32->192(1x1)->25(1x1)->32
test set: x4 case, DIV2K val dataset(801-900), psnr EDSR baseline 28.84, wdsr_a 28.77, wdsr_b 28.37
psnr computing method, SKImage compare_psnr

**Comment by JiahuiYu** [comment]
One of our finding is to set the identity pathway at least S*S*3, where 3 represents for RGB, S represents SR scaling. In this case, 4*4*3 = 48.

Can you try to run quick experiments with resBlocks 8 or fewer, baseline EDSR 96 channels, and its correspondent WDSR-A, WDSR-B?

Meanwhile, @ychfan , do you remember our related experiments regarding this question?

**Comment by suke27** [comment]
@JiahuiYu, correct, the upsample layer just set as you said SxSx3, seems in wdsr paper, 4 Experimental Results, just compare 8 blocks in x2 case. do you have x3 or x4 experimental result. I just want to confirm my test result.

**Comment by JiahuiYu** [comment]
@suke27 I meant to set residual identity pathway SxSx3 and re-run a quick experiment. Your previous experiment has 32->192(1x1)->25(1x1)->32, which is too thin for x4 SR. It was one year ago when we ran x3 and x4 experiments, our result was that WDSR-B is better than WDSR-A, both of which is better than EDSR on x3, x4 and x8.

**Comment by suke27** [comment]
if my experiment is slim for x4, could you share what is your resblock for x4 case.

**Comment by JiahuiYu** [comment]
resBlocks 8, baseline EDSR 96 channels, and its correspondent WDSR-A, WDSR-B, as I ready tell you in a earlier reply...

**Comment by JiahuiYu** [comment]
WDSR-B with 48->48x6(1x1)->48x0.8->48, and EDSR with 96->96. Please double check the #parameters are fair.

By the way, which backbone you are using for training? Can you reproduce our results in x2SR?

**Comment by suke27** [comment]
Thank you for your answer, and happy Chinese new year to you! I will try, by the way, do you have -DIV2k- dataset (realistic training dataset)?, I want to try in track 2,3,4 training dataset

**Comment by JiahuiYu** [comment]
Hi I don’t think I have permission to distribute their dataset. Please ask NTIRE for help. You can also have a look on colab competition page and see if you can get download link.
